### PR TITLE
Sogar registry inside devolutions-gateway

### PR DESCRIPTION
**Pull request by AnastasiiaRomaniuk**

### CODE REVIEW

File: devolutions-gateway/Cargo.toml
Comment (AnastasiiaRomaniuk) [inline devolutions-gateway/Cargo.toml:61]: this can be changed to master after this branch in sogar will be merged

File: devolutions-gateway/src/http/middlewares/sogar_auth.rs
Comment (AnastasiiaRomaniuk) [inline devolutions-gateway/src/http/middlewares/sogar_auth.rs:50]: in other requests the auth header is checked to verify that user have access to this action. In order to be able to play video files in the browser, the header can be missed for the get requests. (but if header will be present it will be checked)

File: devolutions-gateway/src/main.rs
Comment (AnastasiiaRomaniuk) [inline devolutions-gateway/src/main.rs:46]: This was the clippy warning so I fix it.

File: devolutions-gateway/src/http/controllers/sogar_token.rs
Comment (CBenoit) [inline devolutions-gateway/src/http/controllers/sogar_token.rs:33]: Why `clone`?

File: devolutions-gateway/src/http/controllers/sogar_token.rs
Comment (CBenoit) [inline devolutions-gateway/src/http/controllers/sogar_token.rs:46]: You don't need that `clone`
```suggestion
                for user in &self.config.sogar_user {
                    if let (Some(username), Some(hashed_password)) = (&user.username, &user.password) {
                        if username == &username_out {
                            let matched = argon2::verify_encoded(hashed_password.as_str(), password_out.as_bytes());
                            if matched.is_err() || !matched.unwrap() {
                                return (StatusCode::UNAUTHORIZED, None);
                            }

                            return create_token(&self.config.delegation_private_key, user);
```

File: devolutions-gateway/src/http/http_server.rs
Comment (CBenoit) [inline devolutions-gateway/src/http/http_server.rs:44]: Move that before the LogMiddleware (middlewares are executed in LIFO order)

**Review by CBenoit — CHANGES_REQUESTED** [review]
I skimmed through the code, here are a few suggestions.
I'll try to setup that and test on my side as well.

Also fix clippy warnings: `cargo clippy`

File: devolutions-gateway/src/plugin_manager/registry.rs
Comment (CBenoit) [inline devolutions-gateway/src/plugin_manager/registry.rs:1]: I feel this file does not belong to this module because this isn't a plugin. Same goes for `push_files.rs`.

**Review by CBenoit — CHANGES_REQUESTED** [review]

File: devolutions-gateway/src/config.rs
Comment (CBenoit) [inline devolutions-gateway/src/config.rs:102]: A single user can't have both pull AND push permissions?

File: devolutions-gateway/src/http/middlewares/sogar_auth.rs
Comment (AnastasiiaRomaniuk) [reply devolutions-gateway/src/http/middlewares/sogar_auth.rs]: done

File: devolutions-gateway/src/plugin_manager/push_files.rs
Comment (AnastasiiaRomaniuk) [reply devolutions-gateway/src/plugin_manager/push_files.rs]: done

File: devolutions-gateway/src/plugin_manager/push_files.rs
Comment (AnastasiiaRomaniuk) [reply devolutions-gateway/src/plugin_manager/push_files.rs]: done

File: devolutions-gateway/src/plugin_manager/push_files.rs
Comment (AnastasiiaRomaniuk) [reply devolutions-gateway/src/plugin_manager/push_files.rs]: done

File: devolutions-gateway/src/jet_client.rs
Comment (AnastasiiaRomaniuk) [reply devolutions-gateway/src/jet_client.rs]: done

File: devolutions-gateway/Cargo.toml
Comment (AnastasiiaRomaniuk) [reply devolutions-gateway/Cargo.toml:63]: rust-argon2 is used to match the password that comes from user with the one that is located in the gateway.json. The file will contain the hashed password. The password verification is located inside sogar_token.rs get_token method.

File: devolutions-gateway/src/config.rs
Comment (AnastasiiaRomaniuk) [reply devolutions-gateway/src/config.rs:263]: done

File: devolutions-gateway/src/http/controllers/sogar_token.rs
Comment (AnastasiiaRomaniuk) [reply devolutions-gateway/src/http/controllers/sogar_token.rs]: removed

File: devolutions-gateway/src/http/controllers/sogar_token.rs
Comment (AnastasiiaRomaniuk) [reply devolutions-gateway/src/http/controllers/sogar_token.rs]: done

File: devolutions-gateway/src/http/http_server.rs
Comment (AnastasiiaRomaniuk) [reply devolutions-gateway/src/http/http_server.rs:43]: done

File: devolutions-gateway/src/plugin_manager/registry.rs
Comment (AnastasiiaRomaniuk) [reply devolutions-gateway/src/plugin_manager/registry.rs]: removed it to the registry directory. The registry file is now called mod.rs

File: devolutions-gateway/src/config.rs
Comment (AnastasiiaRomaniuk) [reply devolutions-gateway/src/config.rs:102]: When we were discussing the feature with @awakecoding there was a point that we don't need the "admin" user to manage the files. The users should have scoped permissions. That's why the user will only have the pull or push permission.

File: devolutions-gateway/src/plugin_manager/registry.rs
Comment (CBenoit) [reply devolutions-gateway/src/plugin_manager/registry.rs]: Great!

File: devolutions-gateway/src/config.rs
Comment (CBenoit) [reply devolutions-gateway/src/config.rs:102]: Sounds good!

**Review by CBenoit — APPROVED** [review]
LGTM

**Pull request merged**
